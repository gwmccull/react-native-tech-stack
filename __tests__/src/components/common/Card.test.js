import React from 'react';
import renderer from 'react-test-renderer';

import { Card } from '../../../../src/components/common/Card';

describe('Card', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Card />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children', () => {
    class Test extends React.Component {
      render() {
        return React.createElement('Test');
      }
    }

    const tree = renderer.create(
      <Card>
        <Test />
      </Card>
    );
    expect(tree).toMatchSnapshot();
  });
});
