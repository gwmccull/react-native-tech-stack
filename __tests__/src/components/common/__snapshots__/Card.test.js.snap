// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders children 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 2,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "paddingLeft": 4,
      "paddingRight": 4,
    }
  }
>
  <Test />
</View>
`;

exports[`Card renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 2,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "paddingLeft": 4,
      "paddingRight": 4,
    }
  }
/>
`;
